packages/core/src/api/lookup.ts

```diff
@@ -163,6 +189,46 @@ export default class Lookup {
     ]);
   }
 
+  fuzzy<T>(
+    query: string,
+    items: T[],
+    key: keyof T,
+    opts?: {
+      prefix?: string;
+      suffix?: string;
+      /**
+       * If true, only items that match the query will be returned
+       */
+      matchOnly?: boolean;
+    }
+  ): T[] {
+    if (query === "") return items;
+
+    const fuzzied: (T | undefined)[] = [];
```

Comment:
It should only be `T[]`. You can filter out `undefined` items in the loop below.